include/libsemigroups/to-froidure-pin.hpp

```diff
@@ -20,194 +20,261 @@
 #define LIBSEMIGROUPS_TO_FROIDURE_PIN_HPP_
 
 #include <cstddef>      // for size_t
+#include <memory>       // for unique_ptr
 #include <type_traits>  // for enable_if_t, is_base_of
 
```

Comment:
Should there be `#include cong-class.hpp` here too? My initial though is yes, as later we call `cong.run()` so a forward decl isn't sufficient (at least this is what I think `pybind11` was complaining about). If yes, the problem is that we now have a circular include:
`to-froidure-pin.hpp` includes `cong-class.hpp` includes `to-todd-coxeter.hpp` includes `to-froidure-pin.hpp`.

As a test, I hackily tried to remove the `to-todd-coxeter.hpp`'s depended on `to-froidure-pin.hpp` which seemed to make `pybind11` happy. Not sure what the right thing to do is.